collection-loader.js

```diff
@@ -53,12 +68,16 @@ function updateItemsInPlace(
         []
       );
       collection.items = get(collectionSlugToCollection, [slugWithIndex, "items"], []);
-
       if (nestedCollectionLimit && collection.items) {
+        const customLayout = customLayouts[index];
         collection.items.forEach((item, index) => {
-          if (item.type === "collection" && nestedCollectionLimit[get(collection, ["associated-metadata", "layout"])]) {
-            item.childCollectionLimit =
-              nestedCollectionLimit[get(collection, ["associated-metadata", "layout"])][index];
+          if (item.type === "collection") {
```

Comment:
not needed, already checking `const collections = items.filter(item => item && item.type == "collection");`